sympy/sets/sets.py

```diff
@@ -1836,14 +1836,17 @@ def is_function_invertible_in_set(func, setv):
     # Functions known to always be invertible:
     if func in (exp, log):
         return True
-    u = Dummy("u")
-    fdiff = func(u).diff(u)
+    fdiff = func.diff(sym)
     # monotonous functions:
-    # TODO: check subsets (`func` in `setv`)
-    if (fdiff > 0) == True or (fdiff < 0) == True:
+    if fdiff.diff(sym) == 0:
         return True
-    # TODO: support more
-    return None
+    range_in_setv = imageset(sym, fdiff, setv)
+    initial = range_in_setv.start
+    end = range_in_setv.end
+    if initial*end > 0:
```

Comment:
?????????????

Reply:
Here, initial is the minimum value of  f'(x) when x lies in setv and end is the maximum value of f'(x). Now, if initial*end > 0 then it means that f'(x) was of same sign throughout the interval setv and hence, f(x) was monotonic and therefore it will be invertible.